### PR TITLE
Optimize by allowing remove_previous_default to run only when default_tax is changed

### PR DESCRIPTION
**Pull request by nishant-cyro**

### CODE REVIEW

File: core/app/models/spree/zone.rb [hunk 1]
```diff
@@ -14,8 +14,10 @@ class Zone < Spree::Base
 
     validates :name, presence: true, uniqueness: { allow_blank: true }
 
+    scope :having_default_tax, -> { where(default_tax: true) }
```
Comment (jitendra) [inline core/app/models/spree/zone.rb]: Please rename it to with_default_tax

File: core/spec/models/spree/zone_spec.rb [hunk 2]
```diff
@@ -14,6 +14,38 @@
...
+          is_expected.to_not include(zone_not_having_default_tax)
+        end
+      end
+    end
+
+    describe 'callbacks' do
+      it { is_expected.to callback(:remove_previous_default).after(:save).if(:default_tax?).if(:default_tax_changed?) }
+    end
+
+    describe '#remove_previous_default' do
```
Comment (jitendra) [inline core/spec/models/spree/zone_spec.rb]: I think this describe block should be nested under callbacks describe block

**Comment by nishant-cyro** [comment]
@jitendra - done with the changes.

**Comment by jitendra** [comment]
:+1: looks good. Please create direct PR

**Pull request closed without merging**